packages/next/src/middleware/request.ts

```diff
@@ -83,7 +83,7 @@ export class MiddlewareRequest extends Request {
   }
 
   get nextUrl() {
-    return this.nextRequest.url
```

Comment:
Should we look to add some unit tests around the code here? I'm wondering if we'd have caught this bug earlier if we had some.

Reply:
Yeah I'll take a look at doing so, but going to move this into 'ready for review' to start getting some 👀 on this